go/vt/vttablet/tabletserver/throttle/throttler.go

```diff
@@ -1015,7 +1031,29 @@ func (throttler *Throttler) checkStore(ctx context.Context, appName string, stor
 	if !throttler.IsEnabled() {
 		return okMetricCheckResult
 	}
-	return throttler.check.Check(ctx, appName, "mysql", storeName, remoteAddr, flags)
+	if throttlerapp.ExemptFromChecks(appName) {
+		// Some apps are exempt from checks. They are always responded with OK. This is because those apps are
+		// continuous and do not generate a substantial load.
+		return okMetricCheckResult
+	}
+	if !flags.SkipRequestHeartbeats && appName != throttlerapp.VitessName {
+		go throttler.heartbeatWriter.RequestHeartbeats()
+		// This check was made by someone other than the throttler itself, i.e. this came from online-ddl or vreplication or other.
+		// We mark the fact that someone just made a check. If this is a REPLICA or RDONLY tables, this will be reported back
+		// to the PRIMARY so that it knows it must renew the heartbeat lease.
+		atomic.StoreInt64(&throttler.recentCheckValue, 1+atomic.LoadInt64(&throttler.recentCheckTickerValue))
```

Comment:
Ah, so we add 1 to it. I think this at least largely addresses my earlier noted concern around timing.